src/column-layout/grid-column-layout.tsx

```diff
@@ -11,12 +11,16 @@ import { InternalColumnLayoutProps } from './interfaces';
 import { COLUMN_TRIGGERS, ColumnLayoutBreakpoint } from './internal';
 import styles from './styles.css.js';
 
-const COLUMN_DEFS: Record<number, GridProps.ElementDefinition | undefined> = {
-  1: { colspan: { default: 12, xxs: 12, xs: 12 } },
-  2: { colspan: { default: 12, xxs: 6, xs: 6 } },
-  3: { colspan: { default: 12, xxs: 6, xs: 4 } },
-  4: { colspan: { default: 12, xxs: 6, xs: 3 } },
-};
+const COLUMN_DEFS = new Map<number, GridProps.ElementDefinition>([
+  [1, { colspan: { default: 12, xxs: 12, xs: 12 } }],
+  [2, { colspan: { default: 12, xxs: 6, xs: 6 } }],
+  [3, { colspan: { default: 12, xxs: 6, xs: 4 } }],
+  [4, { colspan: { default: 12, xxs: 6, xs: 3 } }],
+]);
+
+function getColumnDefinition(columns: number) {
+  return (typeof columns === 'number' ? COLUMN_DEFS.get(columns) : COLUMN_DEFS.get(parseInt(columns))) ?? {};
+}
```

Comment:
This change requires additional discussion with the team. Let's skip column layout in this one and open a separate PR later